randovania/games/am2r/gui/preset_settings/am2r_goal_tab.py

```diff
@@ -0,0 +1,89 @@
+import dataclasses
+from typing import Callable
+
+from PySide6 import QtCore
+
+from randovania.game_description.game_description import GameDescription
+from randovania.games.am2r.layout.am2r_configuration import AM2RConfiguration, AM2RArtifactConfig
+from randovania.gui.generated.preset_am2r_goal_ui import Ui_PresetAM2RGoal
+from randovania.gui.lib import signal_handling
+from randovania.gui.lib.window_manager import WindowManager
+from randovania.gui.preset_settings.preset_tab import PresetTab
+from randovania.interface_common.preset_editor import PresetEditor
+from randovania.layout.preset import Preset
+
+
+class PresetAM2RGoal(PresetTab, Ui_PresetAM2RGoal):
+
+    def __init__(self, editor: PresetEditor, game_description: GameDescription, window_manager: WindowManager):
+        super().__init__(editor, game_description, window_manager)
+        self.setupUi(self)
+
+        self.goal_layout.setAlignment(QtCore.Qt.AlignTop)
+        signal_handling.on_checked(self.prefer_metroids_check, self._on_prefer_metroids)
+        signal_handling.on_checked(self.prefer_bosses_check, self._on_prefer_bosses)
+        self.dna_slider.valueChanged.connect(self._on_dna_slider_changed)
+
+    @classmethod
+    def tab_title(cls) -> str:
+        return "Goal"
+
+    @classmethod
+    def uses_patches_tab(cls) -> bool:
+        return False
+
+    def _update_slider_max(self):
+        self.dna_slider.setMaximum(self.num_preferred_locations)
+        self.dna_slider.setEnabled(self.num_preferred_locations > 0)
+
+    def _edit_config(self, call: Callable[[AM2RArtifactConfig], AM2RArtifactConfig]):
+        config = self._editor.configuration
+        assert isinstance(config, AM2RConfiguration)
+
+        with self._editor as editor:
+            editor.set_configuration_field(
+                "artifacts",
+                call(config.artifacts)
+            )
+
+    @property
+    def num_preferred_locations(self) -> int:
+        preferred = 0
+        if self.prefer_metroids_check.isChecked():
+            preferred += 46
+        if self.prefer_bosses_check.isChecked():
+            preferred += 6
+        if preferred > 46:
+            preferred = 46
+        return preferred
```

Comment:
That's some weird logic. You always start with 0. If the first condition is true, you will always have 46 and second condition will always make it 6 or greater than 46, which will than be reset to 46.
Maybe the following (???):
```
    @property
    def num_preferred_locations(self) -> int:
        if self.prefer_metroids_check.isChecked():
           return 46
        if self.prefer_bosses_check.isChecked():
            return 6
        return 0
```

Reply:
Indeed is much cleaner, thanks